src/lib/mongoBackend/mongoRegistrationGet.cpp

```diff
@@ -84,11 +84,20 @@ static void setProvider(ngsiv2::Registration* regP, const mongo::BSONObj& r)
   regP->provider.http.url = (r.hasField(REG_PROVIDING_APPLICATION))? getStringFieldF(r, REG_PROVIDING_APPLICATION): "";
 
   //
-  // FIXME P4: for the moment supportedForwardingMode and legacyForwardingMode are hardwired (i.e. DB is not taken
-  // into account for them)
+  // FIXME #3106: for the moment supportedForwardingMode is hardwired (i.e. DB is not taken
+  // into account for them).
   //
   regP->provider.supportedForwardingMode = ngsiv2::ForwardAll;
-  regP->provider.legacyForwardingMode = true;
+
+  std::string format = r.hasField(REG_FORMAT)? getStringFieldF(r, REG_FORMAT) : "JSON";
+  if (format == "JSON")
```

Comment:
Same doubt here. What about "legacy".
I am probably wrong ...

Reply:
Already answered in the other comment.

Reply:
NTC